src/QueryM/DatumCacheWsp.purs

```diff
@@ -104,24 +106,56 @@ instance DecodeAeson GetDatumsByHashesR where
   decodeAeson r =
     let
       decodeDatumArray
-        :: Aeson -> Either JsonDecodeError (Map DataHash Datum)
+        :: Aeson -> Either JsonDecodeError (Map DataHash (Either String Datum))
       decodeDatumArray =
         caseAesonArray (Left $ TypeMismatch "expected array")
           $ (map Map.fromFoldable) <<< traverse decodeDatum
 
+      decodeEmptyDatums obj =
+        pure Map.empty *>
+          ( caseAesonArray (Left $ TypeMismatch "expected array")
+              $ identity <<< traverse decodeEmptyDatum
+          ) obj
+
+      decodeEmptyDatum
+        :: Aeson -> Either JsonDecodeError Unit
+      decodeEmptyDatum aes =
+        if stringifyAeson aes == "{}" then
+          pure unit
+        else
+          Left $ TypeMismatch "expected empty JSON"
+
       decodeDatum
-        :: Aeson -> Either JsonDecodeError (DataHash /\ Datum)
-      decodeDatum = caseAesonObject (Left $ TypeMismatch "expected object")
-        $ \o -> (/\) <$> map wrap (o .: "hash") <*>
-            (decodeAeson =<< o .: "value")
+        :: Aeson -> Either JsonDecodeError (DataHash /\ Either String Datum)
+      decodeDatum obj = caseAesonObject (Left $ TypeMismatch "expected object")
+        (\o -> (/\) <$> map wrap (o .: "hash") <*> (decodeValueOption obj))
+        obj
+
+      decodeValueOption
+        :: Aeson -> Either JsonDecodeError (Either String Datum)
+      decodeValueOption aes =
+        do
+          options <- (pure <$> getNestedAeson aes [ "value", "Right" ])
+            <|> (Left <$> getNestedAeson aes [ "value", "Left" ])
+          case options of
+            Left x -> pure $ Left $ show x
```

Comment:
This is the place to reproduce the possible error responses from ODC if we want to use a type instead of a String.